File: ai25btech11014/Matgeo/matgeo2.10.3/Codes/vector.py
import numpy as np

P = np.array([1, -1, 2])
Q = np.array([2, 0, -1])
R = np.array([0, 2, 1])

PQ = Q - P
PR = R - P
N = np.cross(PQ, PR)
unit = N / np.linalg.norm(N)

print("Unit vector:", unit)

File: ee25btech11027/MATGEO/2.10.56/codes/vector.c
#include <stdio.h>
#include <math.h>

// Dot product of two 2D vectors
double dot(double a[], double b[]) {
    return a[0]*b[0] + a[1]*b[1];
}

// Magnitude of a 2D vector
double magnitude(double a[]) {
    return sqrt(dot(a, a));
}

// Compute max length M and unit vector u using matrix method
void compute(double a[], double b[], double *M, double u[]) {
    double c = dot(a, b);        // a · b
    *M = sqrt(1 + c);            // largest eigenvalue's sqrt

    // Direction = a + b
    double temp[2] = {a[0] + b[0], a[1] + b[1]};
    double norm = magnitude(temp);
    u[0] = temp[0] / norm;
    u[1] = temp[1] / norm;
}
